src_c/alphablit.c

```diff
@@ -3062,3 +3120,97 @@ pygame_AlphaBlit(SDL_Surface *src, SDL_Rect *srcrect, SDL_Surface *dst,
 {
     return pygame_Blit(src, srcrect, dst, dstrect, the_args);
 }
+
+int
+pygame_Opaque_Blit(SDL_Surface *src, SDL_Rect *srcrect, SDL_Surface *dst,
+                   SDL_Rect *dstrect, int the_args)
+{
+    SDL_Rect fulldst;
+    int srcx, srcy, w, h;
+
+    /* Make sure the surfaces aren't locked */
+    if (!src || !dst) {
+        SDL_SetError("pygame_Blit: passed a NULL surface");
+        return (-1);
+    }
+    if (src->locked || dst->locked) {
+        SDL_SetError("pygame_Blit: Surfaces must not be locked during blit");
+        return (-1);
+    }
+
+    /* If the destination rectangle is NULL, use the entire dest surface */
+    if (dstrect == NULL) {
+        fulldst.x = fulldst.y = 0;
+        dstrect = &fulldst;
+    }
+
+    /* clip the source rectangle to the source surface */
+    if (srcrect) {
+        int maxw, maxh;
+
+        srcx = srcrect->x;
+        w = srcrect->w;
+        if (srcx < 0) {
+            w += srcx;
+            dstrect->x -= srcx;
+            srcx = 0;
+        }
+        maxw = src->w - srcx;
+        if (maxw < w)
+            w = maxw;
+
+        srcy = srcrect->y;
+        h = srcrect->h;
+        if (srcy < 0) {
+            h += srcy;
+            dstrect->y -= srcy;
+            srcy = 0;
+        }
+        maxh = src->h - srcy;
+        if (maxh < h)
+            h = maxh;
+    }
+    else {
+        srcx = srcy = 0;
+        w = src->w;
+        h = src->h;
+    }
+
+    /* clip the destination rectangle against the clip rectangle */
+    {
```

Comment:
is there a missing if statement here?

Reply:
Don't think so, the scope block doesn't really do anything - probably a style thing from whoever originally wrote this code? 

I just copy pasted it all over from the normal `pygame_Blit()` function - one of the things I'd want to fix in a final version of this.